packages/adblocker/src/engine/engine.ts

```diff
@@ -828,6 +917,38 @@ export default class FilterEngine extends EventEmitter<
       isFilterExcluded: this.isFilterExcluded.bind(this),
     });
 
+    if (this.hasListeners()) {
+      const context: CosmeticFilterMatchingContext = {
+        url,
+        domain,
+        hostname,
+
+        classes,
+        hrefs,
+        ids,
+
+        allowGenericHides,
+        allowSpecificHides,
+
+        getBaseRules,
+        getInjectionRules,
+        getExtendedRules,
+        getRulesFromDOM,
+        getRulesFromHostname,
+
+        matchType: FilterType.COSMETIC,
+        reference,
+      };
+
+      for (const match of matches) {
+        this.emit('filter-matched', match, context);
+
+        if (exceptionMatches.has(match)) {
+          this.emit('filter-matched', exceptionMatches.get(match)!, context);
+        }
```

Comment:
Nit. We could do a single look-up in `exceptionMatches` instead of two (there is a similar comment above with an example of how it could look like)

Reply:
Updated.